### PR TITLE
BUG: fix CubicTriInterpolator when masked triangulation is used

### PR DESCRIPTION
**Pull request by godsic**
## PR Summary

If triangulation is masked, there is an assignment of arrays of different shapes in CubicTriInterpolator. It comes down to a typo, i.e., the mask is applied after the assignment.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by GBillotey** [comment]
Related to issue [#11820](https://github.com/matplotlib/matplotlib/issues/11820), thanks for your very concise solution.

**Comment by godsic** [comment]
@jklymak #11820 indeed. Sure, I can take care of the test. It will probably take some time for me to figure out how matplotlib unit test system works.

**Comment by godsic** [comment]
@GBillotey @jklymak I've made some tests based on #11820 example. Didn't manage to test locally, since pytest kept complaining about missing ft2font.

File: lib/matplotlib/tri/triinterpolate.py [hunk 1]
```diff
@@ -408,8 +408,8 @@ def __init__(self, triangulation, z, kind='min_E', trifinder=None,
         self._tri_renum = tri_renum
         # Taking into account the node renumbering in self._z:
         node_mask = (node_renum == -1)
-        self._z[node_renum[~node_mask]] = self._z
         self._z = self._z[~node_mask]
+        self._z[node_renum[~node_mask]] = self._z
```
Comment (timhoffm) [inline lib/matplotlib/tri/triinterpolate.py]: The three lines can written more clearly as
~~~
valid_mask = (node_renum != -1)
self._z[node_renum[valid_mask]] = self._z[valid_mask]
~~~

**Comment by GBillotey** [comment]
The error occurs when a `CubicTriInterpolator` is built on a masked `Triangulation`, in case the said-mask masks out all the triangles of at least one node - this results in the node being 'deleted' and node renumbering occurs internally. (The error is also numpy-version specific, as I get error with 1.15 but not with 1.11).
I believe a simpler test could be proposed based on this, and the correctness of the interpolation could also be tested (testing for instance a linear variation of z).

**Comment by QuLogic** [comment]
Thanks for the PR @godsic!

I've gone ahead and simplified the test and masking in #17358, which replaces this PR.

**Pull request closed without merging**